edith-step/src/components/ReceiptInput.js

```diff
@@ -1,116 +1,230 @@
 import React from 'react';
 import axios from 'axios';
+import createReactClass from 'create-react-class';
+import PropTypes from 'prop-types';
+import 'regenerator-runtime/runtime';
 
+/**
+ * Displays form that takes in user input for a dynamic number
+ * of grocery items and their prices and quantities and adds them
+ * to a list.
+ */
 export default class ReceiptInput extends React.Component {
+  /**
+   * Constructor
+   * @param {Props} props Setup state.
+   */
   constructor(props) {
     super(props);
     this.state = {items: [], itemName: '', itemPrice: 0.0, itemQuantity: 1};
     this.handleChange = this.handleChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
   }
 
-  handleSubmit(e) {
+  /**
+   * Sends item info to the receipt-deals servlet and
+   * responds with the cheapest store for that item.
+   * @param {String} name item name
+   * @param {double} price item price
+   * @param {number} quantity item quantity
+   * @return {Object} newDeal cheapest item
+   */
+  async getDeal(name, price, quantity) {
+    const response = await axios({
+      method: 'post',
+      url: '/receipt-deals',
+      data: {
+        itemName: name,
+        itemPrice: price,
+        itemQuantity: quantity,
+      },
+    });
+    const dealItem = response.data;
+
+    const newDeal = dealItem === 'no deal found' ?
```

Comment:
It would be ideal to have an enum returned for the list of possible stores with one of them being NO_STORE. 
pseudocode:
```
enum {
  TARGET,
  WALMART,
  NO_STORE
}
```

The reason I worry about comparing to 'no deal found' is that if there's accidentally an extra space or something spelled incorrectly then this condition will make it seem like an actual store was found.

The tradeoff is that we would need to update the enum for every new store we add, but for our purposes I think that's ok. Lmk what you think and ofc that can come in a follow-up CL.

Reply:
So would this go in the java servlet? Like the response to the post request would use the enum so that the choices we have in the React file are already set